app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,350 @@
+    derivationScheme,
+    config.shouldExportPubKeyBulk && isFeatureSupported(CryptoProviderFeature.BULK_EXPORT),
+    async (derivationPaths: BIP32Path[]) => {
+      return await withDevice(async (bitbox02: BitBox02API) => {
+        const xpubs: Uint8Array[] = await bitbox02.cardanoXPubs(derivationPaths)
+        return xpubs.map(Buffer.from)
+      })
+    },
+    isFeatureSupported(CryptoProviderFeature.BYRON)
+  )
+
+  function isFeatureSupported(feature: CryptoProviderFeature): boolean {
+    switch (feature) {
+      case CryptoProviderFeature.BYRON:
+      case CryptoProviderFeature.POOL_OWNER:
+      case CryptoProviderFeature.MULTI_ASSET:
+      case CryptoProviderFeature.VOTING:
+        return false
+      default:
+        return BITBOX02_VERSIONS[feature]
+          ? hasRequiredVersion(version, BITBOX02_VERSIONS[feature])
+          : hasRequiredVersion(version, BITBOX02_VERSIONS[CryptoProviderFeature.MINIMAL])
+    }
+  }
+
+  function ensureFeatureIsSupported(feature: CryptoProviderFeature): void {
+    if (!isFeatureSupported(feature)) {
+      throw new InternalError(
+        BITBOX02_ERRORS[feature] ?? BITBOX02_ERRORS[CryptoProviderFeature.MINIMAL],
+        {message: `${version.major}.${version.minor}.${version.patch}`}
+      )
+    }
+  }
+
+  function getHdPassphrase(): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Operation not supported',
+    })
+  }
+
+  function sign(message: HexString, absDerivationPath: BIP32Path): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Operation not supported',
+    })
+  }
+
+  async function displayAddressForPath(
+    absDerivationPath: BIP32Path,
+    stakingPath: BIP32Path
+  ): Promise<void> {
+    await withDevice(async (bitbox02: BitBox02API) => {
+      await bitbox02.cardanoAddress(bb02Network, {
+        pkhSkh: {
+          keypathPayment: absDerivationPath,
+          keypathStake: stakingPath,
+        },
+      })
+    })
+  }
+
+  function getWalletSecret(): void {
+    throw new UnexpectedError(UnexpectedErrorReason.UnsupportedOperationError, {
+      message: 'Unsupported operation!',
+    })
+  }
+
+  function getDerivationScheme(): DerivationScheme {
+    return derivationScheme
+  }
+
+  function prepareInput(input: TxInput, addressToAbsPathMapper: AddressToPathMapper): CardanoInput {
+    return {
+      keypath: addressToAbsPathMapper(input.address),
+      prevOutHash: Buffer.from(input.txHash, 'hex'),
+      prevOutIndex: input.outputIndex,
+    }
+  }
+
+  function prepareOutput(output: TxOutput): CardanoOutput {
+    if (output.tokenBundle.length > 0 && !isFeatureSupported(CryptoProviderFeature.MULTI_ASSET)) {
+      throw new InternalError(InternalErrorReason.BitBox02MultiAssetNotSupported, {
+        message:
+          'Sending tokens is currently not supported on BitBox02. Please wait for the firmware update or use Ledger instead.',
```

Comment:
```suggestion
          'Sending tokens is currently not supported on BitBox02. Please wait for the firmware update or use other wallet options (Ledger, Trezor, Mnemonic) instead.',
```

Reply:
actually, only Ledger seems to be fully interoperable with Bitbox (i.e. same keys/addresses produced from the same mnemonic), that's why I mentioned explicitly only Ledger. I'd expect it to be elaborated on in the bitbox tutorial for Adalite that's being prepared

Reply:
Oh. In that case I would specify this compatibility somehow, e.g. `or use Ledger (with the same seed phrase) instead.`